src/melody/inputs.py

```diff
@@ -43,30 +46,35 @@ def __init__(self, name, options, state=False):
 
     @property
     def state(self):
-        ''' '''
+        '''Returns the state variable. This is False by default.'''
```

Comment:
How can a state be `False`? Should this be named something like `requires_state` instead or, if it is intended to hold a state object, maybe it should be None by default? (None is 'falsey' in python anyway so we keep that functionality if it is used anywhere.)